src/utils/FiltersCategoryInterface.ts

```diff
@@ -11,12 +13,22 @@ export interface CategoryGroup {
   data: FilterItem[];
 }
 
+type CategoryID = keyof FetchQuickstartsOptions;
```

Comment:
As far as I can tell - this is not how things are currently modeled in the quickstarts service.

The `GET /quickstarts` endpoint accepts a list of query parameters, where the query parameters are mapped to the `Tag` model: https://github.com/RedHatInsights/quickstarts/blob/main/pkg/routes/quickstarts.go#L111. The `TagType`'s just so happen to have `product-families`, `use-case`, and `content` here: https://github.com/RedHatInsights/quickstarts/blob/main/pkg/models/tag.go#L16-L18

However  `CategoryID`'s is a separate field on the `Filter` model, and the only three values we have today are `use-case`, `product-families`, and `content`: https://github.com/RedHatInsights/quickstarts/blob/main/pkg/models/filters.go#L31

My concern is nothing in the backend is actually coupling the 2 together and forcing this overlap. 

I'm wondering if these should be 2 separate types versus using the keys from the options type (keeping things closer to how they are modeled on the backend).

Reply:
The CategoryID is only needed on the frontend, so there's no need to couple it with the backend at this point. Additionally, all filters are already defined within their respective categories in the backend.

[](https://github.com/RedHatInsights/quickstarts/blob/main/pkg/models/filters.go#L27)

Reply:
Synced with @JetyAdam offline. I think the way we have this modeled in the UI is correct - however we need to improve the backend to couple `CategoryID` to the `TagType`. `CategoryID` is currently just `string`: https://github.com/RedHatInsights/quickstarts/blob/main/pkg/models/filters.go#L9

Reply:
Here is the link for the Jira ticket: https://issues.redhat.com/browse/RHCLOUD-35500